io_ogre/ogre/scene.py

```diff
@@ -671,15 +671,19 @@ def dot_scene_node_export( ob, path, doc=None, rex=None,
         if ob.data.type == 'SPOT':
             a = doc.createElement('lightRange')
             l.appendChild(a)
-            a.setAttribute('inner',str( ob.data.spot_size * (1.0 - ob.data.spot_blend)))
-            a.setAttribute('outer',str(ob.data.spot_size))
-            a.setAttribute('falloff','1.0')
+            a.setAttribute('inner', str( ob.data.spot_size * (1.0 - ob.data.spot_blend)))
+            a.setAttribute('outer', str(ob.data.spot_size))
+            a.setAttribute('falloff', '1.0')
 
         a = doc.createElement('lightAttenuation'); l.appendChild( a )
-        a.setAttribute('range', '5000' )        # is this an Ogre constant?
-        a.setAttribute('constant', '1.0')       # TODO support quadratic light
-        a.setAttribute('linear', '%6f' % (1.0 / ob.data.distance))
-        a.setAttribute('quadratic', '0.0')
+        # NOTE: No idea why a factor of 10 is required but otherwise it does not look correct
+        light_range = ob.data.cutoff_distance * 10
+        if light_range == 0:
+            light_range = 0.001
+        a.setAttribute('range', light_range)
+        a.setAttribute('constant', '1.0')
+        a.setAttribute('linear', '%6f' % (4.5 / light_range))
+        a.setAttribute('quadratic', '%6f' % (75.0 / (light_range * light_range)))
```

Comment:
this should be `2/distance*distance` if I read the blender docs and this correctly: http://learnwebgl.brown37.net/09_lights/lights_attenuation.html

the other parameters should be 0

Reply:
Thanks, I'll look into it.

Reply:
There is very little difference, but I think that its better using 1/d^2.

- With the formula: 1+4,5 / (10 *d) + 7,5 / (10 * d)^2
![screenshot20240314_223113852](https://github.com/OGRECave/blender2ogre/assets/5248597/b31bf66c-450e-448c-967f-615d3d1a00b3)

- Using the formula proposed in the site (1+ 1 / d^2):
![screenshot20240314_223247667](https://github.com/OGRECave/blender2ogre/assets/5248597/d3a9a937-63bb-438d-a525-ac1b23a5b48f)

- Blender:
![image](https://github.com/OGRECave/blender2ogre/assets/5248597/0fb34bd6-6e87-416f-8298-3bd3944f70ef)

Reply:
there is a much clearer test that you can use:
![grafik](https://github.com/OGRECave/blender2ogre/assets/4941802/39c2c020-0da9-4961-bcaa-418ebcc1d3d9)

Reply:
part of the issue is that `powerScale` in ogre is not the same as `ob.data.energy`:

forcing power scale to 1.0, instead of 1000 (energy) gives you:
![grafik](https://github.com/OGRECave/blender2ogre/assets/4941802/b8b561c7-f371-400f-bb94-3e7afd2120a5)

the remaining issue is the ambient light setting, that ogre-meshviewer currently ignores (which is ok I think) and I think blender does something special at the cutoff distance. a programmatic fade-out I guess. Ogre just does a hard cut.

Reply:
There seems to be some sort of ambient lighting in Blender.

In OGRE I got a hard cut at 5 "meters" which was the range or distance.

In my case I was using shaders from https://learnopengl.com/

But yes, your idea of using a plane is much better.